### PR TITLE
Menuitem, qweb report and BI analysis report

### PR DESCRIPTION
**Pull request by aitorbouzas**
Added new menuitem for grouping in a tree view and printing purposes.

### CODE REVIEW

File: sale_commission/__openerp__.py [hunk 1]
```diff
@@ -41,9 +42,11 @@
         "views/sale_order_view.xml",
         "views/account_invoice_view.xml",
         "views/settlement_view.xml",
+        "views/report_settlement.xml",
         "wizard/wizard_settle.xml",
         "wizard/wizard_invoice.xml",
-        # "report/cc_commission_report.xml"
+        # "report/cc_commission_report.xml",
```
Comment (pedrobaeza) [inline sale_commission/__openerp__.py]: Please remove this line

**Comment by pedrobaeza** [comment]
:+1: (code review), although better to implement this as a BI report view.

File: sale_commission/__openerp__.py (hunk 1 above)
Comment (aitorbouzas) [inline sale_commission/__openerp__.py]: What about the file? Should I remove it too?

**Comment by aitorbouzas** [comment]
I've never made a BI report view. I though that a qweb would be nice since there's already one BI report...

**Comment by pedrobaeza** [comment]
A BI report is a model with _auto=False and an SQL VIEW defined in init method with a graph view associated. Check standard view for sales: https://github.com/odoo/odoo/blob/8.0/addons/sale/report/sale_report.py

**Comment by aitorbouzas** [comment]
Fine, will check it and create one for this PR. Should I leave the qweb also? It's useful for printing the report into pdf files.

**Comment by pedrobaeza** [comment]
Yeah, why not? More options to have.

**Comment by aitorbouzas** [comment]
I see that sale_commission is having a graph in: sale_commission/views/settlement_view.xml#L125
What's is the difference between that and a new model with _auto=False with an associated graph?

**Comment by pedrobaeza** [comment]
The object where you are going to query data. You are asking to have commission data before making settlement, so you have to make a model that contains information extracted from account.invoice.line, and develop their corresponding views.

Settlement graph view is for showing in pivot table and graph data from settlements only.

**Comment by aitorbouzas** [comment]
Almost done, but i'm having an error that I can't fix.
As I click on the graph view, it says:

```
Traceback (most recent call last):
  File "/opt/odoo/openerp/http.py", line 539, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/openerp/http.py", line 576, in dispatch
    result = self._call_function(**self.params)
  File "/opt/odoo/openerp/http.py", line 312, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/openerp/service/model.py", line 118, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/openerp/http.py", line 309, in checked_call
    return self.endpoint(*a, **kw)
  File "/opt/odoo/openerp/http.py", line 805, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/openerp/http.py", line 405, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/addons/web/controllers/main.py", line 944, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/opt/odoo/addons/web/controllers/main.py", line 936, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, *args, **kwargs)
  File "/opt/odoo/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/opt/odoo/openerp/models.py", line 2144, in read_group
    cr.execute(query, where_clause_params)
  File "/opt/odoo/openerp/sql_db.py", line 158, in wrapper
    return f(self, *args, **kwargs)
  File "/opt/odoo/openerp/sql_db.py", line 234, in execute
    res = self._obj.execute(query, params)
ProgrammingError: column sale_commission_analysis_report.id does not exist
LINE 2:             SELECT min(sale_commission_analysis_report.id) A...
```

It tries to get the column "id", but there's no "id" column on the view... Neither it is defined in the sale_report example. How can I achieve this?

**Comment by aitorbouzas** [comment]
Saw it! I just needed another column in the view definition.

**Comment by pedrobaeza** [comment]
Please check Pylint errors: https://travis-ci.org/OCA/commission/jobs/128019311

**Comment by aitorbouzas** [comment]
Working on it!

**Comment by aitorbouzas** [comment]
Seems ok now.

**Comment by aitorbouzas** [comment]
Needs some review and testing.

**Comment by hendrixcosta** [comment]
:+1:

**Comment by rubencabrera** [comment]
Looks good to me, quite useful, these modules need more visualization options and this PR helps!

**Comment by pedrobaeza** [comment]
Please check branch conflicts and your CLA.

**Comment by aitorbouzas** [comment]
I already sent the CLA.
Should I give you write access to my repository to merge it? Or how...?

**Comment by pedrobaeza** [comment]
Hi, Aitor, I'll check the CLA.

About the merge, you have to rebase your branch and resolve conflicts, force pushing again to this branch to get a clean PR.

**Comment by aitorbouzas** [comment]
Can someone throw some light about this error?
Are the tests trying to settle it without paying it before?

"The Type of Commission only allows create the Settlements when the Invoices are Paid."

**Comment by ivantodorovich** [comment]
What's the status of this?
@Softdil can you please rebase this PR's branch?

**Comment by aitorbouzas** [comment]
I already rebased, and then I needed some helped with the descripted error, because I don't know how to solve it...

**Comment by Daniel-CA** [comment]
The test is giving this error:
2016-06-29 15:26:35,619 23848 ERROR openerp_test openerp.addons.sale_commission.tests.test_sale_commission: ` AssertionError: 1 != 0 : The Type of Commission only allows create the Settlements when the Invoices are Paid.

Nothing to do with your code, try rebasing again.

**Comment by pedrobaeza** [comment]
Is this going to be fixed?

**Comment by aitorbouzas** [comment]
Fixed, added multi-company rule also.

**Comment by pedrobaeza** [comment]
Thanks. I'm not using anymore 8.0, so this is not interesting for me. Anyone willing to review this?

**Comment by aitorbouzas** [comment]
Is self reviewing valid in any way? I just tested it and it works nice, there's only one problem but not with this module. Something is wrong when calculating the 5% + 10% special commission in invoice.

**Comment by pedrobaeza** [comment]
@aitorbouzas are you still working on 8.0? I can review it if you are going to forward-port it this to at least 10.0

**Comment by aitorbouzas** [comment]
Sorry, i'm no longer working on 10.0 or lower versions right now. This module was left working correctly.

**Comment by pedrobaeza** [comment]
Are you saying that you are working now on 11.0? It would be great to move this to that version.

**Comment by aitorbouzas** [comment]
Ok, I can move it to version 11.0 if you promise you will review it ;P
P.D: It doesn't need to be migrated to 9/10? Right?

**Comment by pedrobaeza** [comment]
Promised!

**Comment by aitorbouzas** [comment]
Closing as this has been here for 2 years now with no one interested in it.

**Pull request closed without merging**